FreeRTOS-Plus/Source/FreeRTOS-Plus-TCP/FreeRTOS_ARP.c

```diff
@@ -583,7 +605,7 @@ NetworkBufferDescriptor_t *pxNetworkBuffer;
 		if( xIsCallingFromIPTask() != 0 )
 		{
 			/* Only the IP-task is allowed to call this function directly. */
-			xNetworkInterfaceOutput( pxNetworkBuffer, pdTRUE );
+			( void ) xNetworkInterfaceOutput( pxNetworkBuffer, pdTRUE );
```

Comment:
Is it possible to check the return value here?

```xNetworkInterfaceOutput``` is implemented by third party, that I would check the return value to be sure the API contract is met.

Reply:
There is no way to return the error to the caller since the function returns a void. I can go ahead and add a configASSERT which seems drastic to me.
Or, as you suggested I can add a logging instead of 'configASSERT'ing.
@htibosch, any comments on what would be better?

Reply:
Hello, the function `xNetworkInterfaceOutput()` is supposed to return a non-zero value if it was successful. The return value could be logged, but the is no different behaviour of the IP-task: when a packet can not be delivered, that is just bad luck. it will be tried later.

Reply:
Thanks Hein, I have added a looging functionality for easier debugging.

Reply:
The best thing to do would be to count these and have some utility for retrieving counts for intermittent logging (say once per minute or via CLI command).  Having counts of delivered/undelivered packets, etc. is incredibly useful for debugging and can be done without affecting runtime behavior as happens with logging.

However, this is out of scope for this PR.  The purpose of this PR is not to add new capabilities to the code, but to eliminate some Coverity warnings.

Reply:
I agree completely with @gkwicker